src/js/components/CheckBox/propTypes.js

```diff
@@ -13,6 +13,7 @@ if (process.env.NODE_ENV !== 'production') {
     name: PropTypes.string,
     onChange: PropTypes.func,
     pad: padPropType,
+    plain: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
```

Comment:
when would `plain` be a "string"?

Reply:
So in `FormField.js` on line 220 there is 
```
          if (
            child &&
            child.type &&
            grommetInputNames.indexOf(child.type.displayName) !== -1 &&
            child.props.plain === undefined &&
            child.props.focusIndicator === undefined
          ) {
            return cloneElement(child, {
              plain: 'noBorder',
              focusIndicator: false,
            });
          }

```

in this case I changed the `plain: 'noBorder'` instead of true so that way it was not setting plain on `checkbox`. 
setting `plain: 'noBorder'` also triggers TextInput and Select to have their plain be true to get rid of the double border.

Reply:
This PR also is not the only direction depending on which route we want to go:
we have a plain prop and our theme looks specific if plain is true then take off horizontal pad.
Or we can have in our hpe theme
CheckBox: {
    vertical: ‘xsmall’,
    horizontal: ‘small’
},
Then someone would have to pass `pad={{ vertical: ‘xsmall’ }}` as a prop to `checkbox`